### PR TITLE
Stabilize `std::io::Buf{Reader, Writer}::capacity`

### LINKED ISSUE #68833: Tracking issue for `std::io::{BufReader, BufWriter}::capacity`
#68558 adds `capacity` methods to `BufReader` and `BufWriter`.

Unresolved questions:

### PR DESCRIPTION
**Pull request by JohnTitor**
Closes #68833
FCP is done here: https://github.com/rust-lang/rust/issues/68833#issuecomment-637596083

### CODE REVIEW

File: src/libstd/io/buffered.rs [hunk 1]
```diff
@@ -198,7 +197,7 @@ impl<R> BufReader<R> {
     ///     Ok(())
     /// }
     /// ```
-    #[unstable(feature = "buffered_io_capacity", issue = "68833")]
+    #[stable(feature = "buffered_io_capacity", since = "1.46.0")]
```
Comment (JohnTitor) [inline src/libstd/io/buffered.rs:200]: I set `1.46.0` here since promoting will happen soon.

**Comment by Lucretiel** [comment]
Not a blocker, but related: https://github.com/rust-lang/rust/issues/72919

**Comment by shepmaster** [comment]
> I set `1.46.0` here

> @jonas-schievink added this to the 1.45 milestone 18 minutes ago

**Comment by jonas-schievink** [comment]
Milestones don't have to be accurate right away, but they need to be set so the release team can update them when a new Rust version is released.

**Comment by shepmaster** [comment]
Based on the FCP...

@bors r+ rollup

**Comment by JohnTitor** [comment]
Updated the milestone following the stabilize version.

**Pull request merged**